crates/nu-command/src/commands/any.rs

```diff
@@ -52,74 +50,88 @@ impl WholeStreamCommand for Command {
     }
 }
 
-fn any(args: CommandArgs) -> Result<ActionStream, ShellError> {
-    let ctx = Arc::new(EvaluationContext::from_args(&args));
-    let tag = args.call_info.name_tag.clone();
-    let (Arguments { block }, input) = args.process()?;
+fn any(args: CommandArgs) -> Result<OutputStream, ShellError> {
+    let ctx = EvaluationContext::from_args(&args);
+    let mut args = args.evaluate_once()?;
+    let any_args = AnyArgs {
+        predicate: args.req(0)?,
+    };
+
+    let err = Err(ShellError::labeled_error(
+        "Expected a condition",
+        "expected a condition",
+        args.call_info.name_tag.clone(),
+    ));
 
+    //This seems a little odd. Can't we have predicates with pipelines/multiple statements?
     let condition = {
-        if block.block.block.len() != 1 {
-            return Err(ShellError::labeled_error(
-                "Expected a condition",
-                "expected a condition",
-                tag,
-            ));
+        if any_args.predicate.block.block.len() != 1 {
+            return err;
         }
-        match block.block.block[0].pipelines.get(0) {
+        match any_args.predicate.block.block[0].pipelines.get(0) {
             Some(item) => match item.list.get(0) {
                 Some(ClassifiedCommand::Expr(expr)) => expr.clone(),
                 _ => {
-                    return Err(ShellError::labeled_error(
-                        "Expected a condition",
-                        "expected a condition",
-                        tag,
-                    ));
+                    return err;
                 }
             },
             None => {
-                return Err(ShellError::labeled_error(
-                    "Expected a condition",
-                    "expected a condition",
-                    tag,
-                ));
+                return err;
             }
         }
     };
 
-    let cond = Ok(InputStream::one(
-        UntaggedValue::boolean(false).into_value(&tag),
-    ));
+    let scope = args.scope.clone();
+
+    // Variables in nu are immutable. Having the same variable accross invocations
+    // of evaluate_baseline_expr does not mutate the variables and thus each
+    // invocations are independent of each other!
+    scope.enter_scope();
+    scope.add_vars(&any_args.predicate.captured.entries);
+    let result: Result<bool, ShellError> = args.input.try_any(|row| {
```

Comment:
I'm wondering what the advantage of `.try_any` is here over the built-in `.any`?

Reply:
try_any allows for short_circuiting here. (like the other try_x commands in std::iter::Iterator)

Reply:
Yeah, I did figure it out once I saw your updates to the other PR. Some comments here: https://github.com/nushell/nushell/pull/3312#discussion_r615295399